### PR TITLE
Refactor location import to ingest and create necessary objects from provided GeoJSON file

### PR DESCRIPTION
**Pull request by hancush**
## Overview

This PR:

- Removes `import_osm` and `link_locations`, replacing them with a unified `import_locations` command that ingests a provided GeoJSON file and creates the appropriate `Location` objects from it.
    - #704 suggests removing `clean_locations` to remove locations that are unreferenced by other models, but I left it in place because it still seems useful to me, at least to clean up cruft from previous versions of the data.
- Adds an `sfm` attribute to the `Location` model to store new metadata provided with the GeoJSON.
- Removes all location inference from `import_google_docs`, instead relying on explicitly related locations for units and events.
- Deprecates outdated spreadsheet fields and shortcodes (mostly related to location fields containing information redundant with the location table in unit and incident sheets), and updates remaining ones.
- Adds a new GeoJSON fixture and test for the `import_locations` command.
- Updates importer test fixtures and tests to reflect the revised location data structure and use the new location command.

Connects #704 

## Notes

I've indicated brand new code in comments. Those can use close review. I've also tried to provide context for the changes to the import. There are probably places to improve that code further, but we are working on a deadline, so I left most of the structure in tact and focused on revising how we access locations, as not to introduce unrelated bugs.

## Testing Instructions

These changes should be covered by the automated test suite.

### CODE REVIEW

File: api/country_views.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def get_context_data(self, **kwargs):
               tags,
               ST_ASGeoJSON(ST_Simplify(geometry, %s))::json AS geometry
             FROM osm_data
-            WHERE admin_level = 2
+            WHERE admin_level = '2'
```
Comment (hancush) [inline api/country_views.py:28]: TODO: I think this should revert back to an int.

File: area/models.py [hunk 2]
```diff
@@ -52,8 +52,8 @@ class AreaName(ComplexField):
     object_ref = models.ForeignKey('Area')
     value = models.TextField(default=None, blank=True, null=True)
     field_name = _("Area of Operations")
-    shortcode = 'u_an'
-    spreadsheet_field_name = 'unit:area_ops_name'
+    # shortcode = 'u_ln'
```
Comment (hancush) [inline area/models.py:55]: TODO: Should I remove or update these?

Comment (tlongers) [reply area/models.py:55]: Yes, I think we should update the shortcodes. I'll prepare a list for you.

File: api/country_views.py (hunk 1 above)
Comment (hancush) [reply api/country_views.py:28]: This API is deprecated.

File: association/models.py [hunk 3]
```diff
@@ -48,18 +48,18 @@ class AssociationStartDate(ComplexField):
     object_ref = models.ForeignKey('Association')
     value = ApproximateDateField()
     field_name = _("First Cited Date")
-    shortcode = 'u_aofcd'
-    spreadsheet_field_name = 'unit:area_ops_first_cited_date'
+    shortcode = 'u_lfcd'
```
Comment (hancush) [inline association/models.py:51]: Spreadsheet field names and shortcodes should match this sheet: https://docs.google.com/spreadsheets/d/1-U_pVNDWFIlG8jLZLVujmV_FZ0H4IsbtfCeRiM8HeNQ/edit?usp=sharing

File: tests/test_importer.py [hunk 4]
```diff
@@ -105,14 +88,17 @@ def test_sources(data_import, data_folder):
...
             ])
             permitted_person_set = set(['PersonName'])
             permitted_incident_set = set([
                 'ViolationStartDate', 'ViolationStatus', 'ViolationType',
                 'ViolationFirstAllegation', 'ViolationDescription',
                 'ViolationEndDate', 'ViolationLastUpdate',
-                'ViolationPerpetratorClassification'
+                'ViolationPerpetratorClassification',
+                'ViolationLocationDescription',
+                'ViolationPerpetratorOrganization'
```
Comment (hancush) [inline tests/test_importer.py:101]: These were sourced before, so I'm not sure why they weren't in the permitted array initially.

File: tests/test_importer.py [hunk 5]
```diff
@@ -105,14 +88,17 @@ def test_sources(data_import, data_folder):
                 'CompositionChild', 'OrganizationName',
                 'MembershipOrganizationMember', 'MembershipOrganizationOrganization',
                 'MembershipPersonOrganization', 'MembershipPersonMember',
-                'CompositionParent',
+                'CompositionParent', 'EmplacementSite', 'EmplacementStartDate',
+                'EmplacementOrganization'
```
Comment (hancush) [inline tests/test_importer.py:92]: Ditto not sure why these weren't already permitted values.

File: sfm_pc/management/commands/import_locations.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import connection
+from django.db.utils import ProgrammingError
+from django.conf import settings
+
+from location.models import Location
+
+
+class Command(BaseCommand):
```
Comment (hancush) [inline sfm_pc/management/commands/import_locations.py:18]: This is all new code and can use careful review.

File: tests/test_location.py [hunk 7]
```diff
@@ -16,6 +43,44 @@ def expected_entity_names(violation, emplacement):
...
+def locations_with_admin_level(level):
+    return Location.objects.filter(
+        **{'sfm__location:admin_level_{}__isnull'.format(level): False}
+    ).exclude(
+        **{'sfm__location:admin_level': int(level)}
+    )
+
+
+@pytest.mark.django_db(transaction=True)
+def test_location_import(initial_location, location_data_import, location_fixture_data):
```
Comment (hancush) [inline tests/test_location.py:55]: This tests new code and can use careful review.

File: sfm_pc/management/commands/import_google_doc.py [hunk 8]
```diff
@@ -37,8 +37,7 @@
 from organization.models import Organization, OrganizationAlias, \
     OrganizationClassification, OrganizationName, OrganizationRealStart
 
-from sfm_pc.utils import (import_class, get_osm_by_id, get_hierarchy_by_id,
-                          CONFIDENCE_MAP, execute_sql)
+from sfm_pc.utils import (import_class, CONFIDENCE_MAP, execute_sql)
```
Comment (hancush) [inline sfm_pc/management/commands/import_google_doc.py:40]: The changes in this file are primarily concerned with removing complicated location inference code and replacing it with simple retrieval of a related location by its "humane ID". The diff is pretty ugly, but hopefully this helps to frame the changes you see.

Comment (hancush) [reply sfm_pc/management/commands/import_google_doc.py:26]: I also flake8'd it.

File: sfm_pc/management/commands/import_locations.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+            '--location_file',
+            dest='location_file',
+            default='fixtures/locations.geojson',
+            help='Relative path to location file for import'
+        )
+
+    def handle(self, *args, **options):
+
+        self.location_file = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
```
Comment (fgregg) [inline sfm_pc/management/commands/import_locations.py:26]: i think you want to use the current working directory. `os.getcwd()`

File: sfm_pc/management/commands/import_locations.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+              feature_type,
+              division_id,
+              tags,
+              sfm,
+              adminlevel,
+              adminlevel1_id,
+              adminlevel2_id,
+              geometry
+            )
+            SELECT
```
Comment (fgregg) [inline sfm_pc/management/commands/import_locations.py:96]: i'd consider a CTE here to make this a little more self-documenting, but do not require that change

**Review by fgregg — CHANGES_REQUESTED** [review]

File: sfm_pc/management/commands/import_locations.py (hunk 10 above)
Comment (hancush) [reply sfm_pc/management/commands/import_locations.py:99]: Cool, good thought. I gave the joined tables more helpful aliases.

**Review by fgregg — APPROVED** [review]

**Pull request merged**